cip/1.accepted/CIP2017-02-06-Path-Patterns.adoc

```diff
@@ -0,0 +1,503 @@
+  `()-/< 𝜶 >/-()` denotes that the Path Pattern `𝜶` should be interpreted in any direction.
+• A reference to a Named Path Predicate +
+  `()-/~alpha/-()` denotes a reference to a Named Path Predicate `alpha`.
+
+Binding of a relationship to a variable is only allowed in the most simple case of a Path Pattern, where only a single relationship is matched by the pattern.
+For binding a whole path to a variable, Path Assignment should be used, by preceding the path with an identifier and an equals sign (`=`).
+This avoids a problem that existed in the past with repetition of relationships (a syntax that is unsupported as of the introduction of Path Patterns), where a relationship variable would bind to a list, making it hard to express predicates over the actual relationships.
+Predicates on parts of a Path Pattern are instead expressed through the use of explicitly Named Path Predicates.
+
+=== Syntax
+
+Path Patterns are part of the Pattern syntax of Cypher.
+
+[source, ebnf]
+----
+Pattern = PatternPart, {',', PatternPart} ;
+PatternPart = [Variable, '='], NodePattern, {(EdgePattern | PathPattern), NodePattern} ;
+
+NodePattern = '(', [Variable], [NodeLabels], [Properties], ')' ;
+
+EdgePattern = (LeftArrowHead, Dash, [EdgeBody], Dash, RightArrowHead)
+            | (LeftArrowHead, Dash, [EdgeBody], Dash)
+            |                (Dash, [EdgeBody], Dash, RightArrowHead)
+            |                (Dash, [EdgeBody], Dash)
+            ;
+EdgeBody = '[', [Variable], [EdgeLabels], [Properties], ']' ;
+
+PathPattern = (LeftArrowHead, Dash, '/', PathExpression, '/', Dash, RightArrowHead)
+            | (LeftArrowHead, Dash, '/', PathExpression, '/', Dash)
+            |                (Dash, '/', PathExpression, '/', Dash, RightArrowHead)
+            |                (Dash, '/', PathExpression, '/', Dash)
+            ;
+
+PathExpression  = {PathAlternative} ;
+PathAlternative = PathRepetition, {'|', PathRepetition} ;
+PathRepetition  = PathDirection, [('*', [RangeDetail]) | '+' | '?'] ;
+PathDirection   = ['<'], PathBase, ['>'] ;
+PathBase = PathEdge
+         | PathAny
+         | PathNode
+         | PathReference
+         | PathGroup
+         ;
+PathEdge      = ':', LabelName ;
+PathAny       = '-' ;
+PathNode      = '(', [NodeLabels], [Properties], ')' ;
+PathReference = '~', SymbolicName ;
+PathGroup     = '[', PathExpression, [Properties], ']' ;
+
+EdgeLabel  = ':', LabelName ;
+EdgeLabels = ':', LabelName, {'|', [':'], LabelName} ;
+NodeLabels = ':', LabelName, ({':', LabelName} | {'|', LabelName}) ;
+LabelName  = SymbolicName ;
+
+RangeDetail = [IntegerLiteral | Parameter], '..', [IntegerLiteral | Parameter] ;
+----
+
+The `PathReference` is a reference to a Named Path Predicate.
+These are defined using the following syntax:
+
+[source, ebnf]
+----
+NamedPathPredicate   = 'PATH', 'PATTERN', NamedPathName, '=', PathPattern, [Where] ;
```

Comment:
It looks like examples are not match with grammar for ```NamedPathPredicate```. In examples like 
```
PATH PATTERN unreciprocated_love = (a)-[:LOVES]->(b)
```
we see, that it should be something like ```PatternPart``` (maybe without ```[Variable, '=']``` part) in ```NamedPathPredicate``` after ```=``` instead of ```PathPattern```.

So, I think that correct rule should be 
```
NamedPathPredicate   = 'PATH', 'PATTERN', NamedPathName, '=', NodePattern, {(EdgePattern | PathPattern), NodePattern}, [Where] ;
```

Reply:
Yes, you are right, thank you!